trie/database.go

```diff
@@ -753,10 +783,10 @@ func (db *Database) Commit(node common.Hash, report bool, callback func(common.H
 }
 
 // commit is the private locked version of Commit.
-func (db *Database) commit(hash common.Hash, batch ethdb.Batch, uncacher *cleaner, callback func(common.Hash)) error {
-	// If the node does not exist, it's a previously committed node
+func (db *Database) commit(hash common.Hash, batch ethdb.Batch, uncacher ethdb.KeyValueWriter, callback func(common.Hash)) error {
+	// If the node does not exist or marked as committed, then it's a previously committed node
 	node, ok := db.dirties[hash]
-	if !ok {
+	if !ok || node.commited {
```

Comment:
I would add the check here  `if  node == nil`, then return nil and remove check `if n == nil` inside  `forChilds`.

Reply:
Yes, we don't need to check `nil` then I will remove unnecessary check `if n == nil` inside the child and keep this like as it is. the `nil` check is my double check on the interim code but I think we don't need it.